FastSimulation/Calorimetry/src/CalorimetryManager.cc

```diff
@@ -1406,18 +1320,20 @@ void CalorimetryManager::loadMuonSimTracks(edm::SimTrackContainer &muons) const
 	{
 	  muons[i].setTkPosition(itcheck->trackerSurfacePosition());
 	  muons[i].setTkMomentum(itcheck->trackerSurfaceMomentum());
-	  //	  std::cout << " Found the SimTrack " << std::endl;
-	  //	  std::cout << *itcheck << std::endl;
-	  //	  std::cout << "SimTrack Id "<< id << " " << muons[i] << " " << std::endl;
 	}
-      //      else
-      //	{
-      //	  std::cout << " Calorimetery Manager : this should really not happen " << std::endl;
-      //	  std::cout << " Was looking for " << id << " " << muons[i] << std::endl;
-      //	  for(unsigned i=0;i<muonSimTracks.size();++i)
-      //	    std::cout << muonSimTracks[i] << std::endl;
-      //	}
     }
   
 }
 
+
+void CalorimetryManager::harvestMuonSimTracks(edm::SimTrackContainer &c) const
+{
+  c.reserve(muonSimTracks.size()+savedMuonSimTracks.size());
+  for(unsigned i=0; i<muonSimTracks.size(); i++) {
+    if(muonSimTracks[i].momentum().perp2() > 1.0 && fabs(muonSimTracks[i].momentum().eta()) < 3.0)  c.push_back(muonSimTracks[i]);
+  }
+  for(unsigned i=0; i<savedMuonSimTracks.size(); i++) {
+    if(savedMuonSimTracks[i].momentum().perp2() > 1.0 && fabs(savedMuonSimTracks[i].momentum().eta()) < 3.0)  c.push_back(savedMuonSimTracks[i]);
+  }
+}
```

Comment:
@skurz - perhaps, request removal of unused capacity?
```
c.shrink_to_fit();
```

Reply:
Done.